pkg/api/validate.go

```diff
@@ -15,17 +16,36 @@ var (
 		`(\.([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]))*` +
 		`$`)
 
-	rxAgentPoolProfileVNetSubnetID = regexp.MustCompile(`(?i)^` +
+	rxVNetID = regexp.MustCompile(`(?i)^` +
 		`/subscriptions/[^/]+` +
 		`/resourceGroups/[^/]+` +
 		`/providers/Microsoft\.Network` +
 		`/virtualNetworks/[^/]+` +
-		`/subnets/[^/]+` +
 		`$`)
 
 	rxAgentPoolProfileName = regexp.MustCompile(`(?i)^[a-z0-9]{1,12}$`)
 )
 
+var (
+	clusterNetworkCIDR *net.IPNet
```

Comment:
These don't seem appropriate here.  Seems like they should be subbed in other places in configuration (master-startup) as well if we're defining variables for them.  Probably means they should live outside of the validate files.  Likely need to be configuration items anyway as noted below.

Reply:
I'm trying to bound the change because we need to backport.  #204 covers doing this work fully.